src/Language/PureScript/Errors.hs

```diff
@@ -566,6 +568,11 @@ prettyPrintSingleError full level e = prettyPrintErrorMessage <$> onTypesInError
               , line $ "The definition has the following uncovered cases:\n"
               , Box.hsep 1 Box.left (map (paras . map (line . prettyPrintBinderAtom)) (transpose bs))
               ] ++ if not b then [line "..."] else []
+    goSimple (OverlappingPattern bs b) =
+      indent $ paras $ [ line "Redundant cases has been detected."
+              , line $ "The definition has the following redundant cases:\n"
```

Comment:
@paf31 Yes, it is included in `checkExhaustive`, just after the exhaustivity checking.

For example:

```haskell
module Main where
f true = 0
f true = 0
g true = 0
g true = 0
```
raises:
```
Multiple warnings:
  Warning in module Main:
  Warning in value declaration g:
  Warning at /dev/stdin line 4, column 1 - line 5, column 1:
      Pattern could not be determined to cover all cases.
      The definition has the following uncovered cases:

      false
  See https://github.com/purescript/purescript/wiki/Error-Code-NotExhaustivePattern for more information, or to contribute content related to this error.

  Warning in module Main:
  Warning in value declaration g:
  Warning at /dev/stdin line 4, column 1 - line 5, column 1:
      Redundant cases has been detected.
      The definitions has the following redundant cases:

      true
  See https://github.com/purescript/purescript/wiki/Error-Code-OverlappingPattern for more information, or to contribute content related to this error.

  Warning in module Main:
  Warning in value declaration f:
  Warning at /dev/stdin line 2, column 1 - line 3, column 1:
      Pattern could not be determined to cover all cases.
      The definition has the following uncovered cases:

      false
  See https://github.com/purescript/purescript/wiki/Error-Code-NotExhaustivePattern for more information, or to contribute content related to this error.

  Warning in module Main:
  Warning in value declaration f:
  Warning at /dev/stdin line 2, column 1 - line 3, column 1:
      Redundant cases has been detected.
      The definitions has the following redundant cases:

      true
  See https://github.com/purescript/purescript/wiki/Error-Code-OverlappingPattern for more information, or to contribute content related to this error.
```